jovian/tests/utils/test_misc.py

```diff
@@ -0,0 +1,165 @@
+        self.assertEqual(urljoin(base_url_1, path_1), expected_result)
+        self.assertEqual(urljoin(base_url_1, path_2), expected_result)
+        self.assertEqual(urljoin(base_url_2, path_1), expected_result)
+        self.assertEqual(urljoin(base_url_2, path_2), expected_result)
+
+    def test_urljoin_trailing_slash(self):
+        base_url_1 = "https://jovian.ml"
+        base_url_2 = "https://jovian.ml/"
+
+        path_1 = "user/siddhant/"
+        path_2 = "/user/siddhant/"
+
+        expected_result = "https://jovian.ml/user/siddhant/"
+
+        self.assertEqual(urljoin(base_url_1, path_1), expected_result)
+        self.assertEqual(urljoin(base_url_1, path_2), expected_result)
+        self.assertEqual(urljoin(base_url_2, path_1), expected_result)
+        self.assertEqual(urljoin(base_url_2, path_2), expected_result)
+
+    def test_urljoin_multiple(self):
+        base_url_1 = "https://jovian.ml"
+
+        path_1 = "user/siddhant/"
+        path_2 = "/gists/starred"
+
+        expected_result = "https://jovian.ml/user/siddhant/gists/starred"
+
+        self.assertEqual(urljoin(base_url_1, path_1, path_2), expected_result)
+
+    def test_urljoin_malformed(self):
+        base_url_1 = "https://jovian.ml"
+        base_url_2 = "https://jovian.ml/"
+
+        path_1 = "///user/siddhant/"
+        path_2 = "//user/siddhant//////"
+
+        expected_result = "https://jovian.ml/user/siddhant/"
+
+        self.assertEqual(urljoin(base_url_1, path_1), expected_result)
+        self.assertEqual(urljoin(base_url_1, path_2), expected_result)
+        self.assertEqual(urljoin(base_url_2, path_1), expected_result)
+        self.assertEqual(urljoin(base_url_2, path_2), expected_result)
+
+    def test_urljoin_no_args(self):
+        with self.assertRaises(TypeError) as cm:
+            urljoin()
+
+        self.assertEqual(str(cm.exception), "urljoin requires at least one argument")
+
+    def test_urljoin_single_url(self):
+        base_url_1 = "https://jovian.ml"
+        base_url_2 = "https://jovian.ml/"
+
+        self.assertEqual(urljoin(base_url_1), base_url_1)
+        self.assertEqual(urljoin(base_url_2), base_url_2)
+
+    def test_urljoin_single_path(self):
+        path_1 = "///user/siddhant"
+
+        expected_result = "user/siddhant"
+
+        self.assertEqual(urljoin(path_1), expected_result)
+        self.assertEqual(urljoin(path_1), expected_result)
+
+
+class TestTimestampMs(TestCase):
+    @mock.patch('time.time', mock.Mock(return_value=1577112328.529031))
+    def test_timestamp_ms(self):
+        self.assertEqual(timestamp_ms(), 1577112328529)
+
+
+class TestGetFlavor(TestCase):
+    def test_get_flavor(self):
+        self.assertEqual(get_flavor(), "jovian")
+
+    def test_get_flavor_import_error(self):
+        import builtins
+        _original_import = builtins.__import__
+
+        def error_raiser(*args, **kwargs):
+            raise ImportError('fake import error')
+
+        builtins.__import__ = error_raiser
```

Comment:
We can put the import in a `mock.patch` decorator like 
```python
@mock.patch("builtins.__import__", mock.Mock(return_value=ImportError('fake import error'))
```

I've done that in some tests.. what do you think?